### ISSUE TITLE
Chunking files longer than 30 seconds

### ISSUE THREAD

**Issue by brettdavies** [nkilm/offline-whisperx#2]
Hi @nkilm! I found your repo, which does most of what I want. Did you happen to test it using files longer than 30 seconds? I'm attempting to chunk long-form audio into ~30-second segments and want to use a VAD model to avoid fixed-length splitting, which is likely to split a sentence or a word. Is that anything you've played with? If not, I'm happy to do the work if you can point me in the right direction.

The [whisperX](https://github.com/m-bain/whisperX) library supports something similar, but I'm not sure if you implemented it in this wrapper.

🗣️ VAD preprocessing, reduces hallucination & batching with no WER degradation

Kind regards.

**Comment by nkilm** [comment]
Hi Brett!

**Yes** I did test the script against longer-length audios(_~10mins_) it was working fine as expected - I had tested with both transcription and diarization.

> I'm attempting to chunk long-form audio into ~30-second segments and want to use a VAD model to avoid fixed-length splitting,

Is there any particular reason why you want to do this process manually? I guess as per the research paper [WhisperX: Time-Accurate Speech Transcription of Long-Form Audio](https://arxiv.org/abs/2303.00747) the default chunk size is `30`.

**Comment by brettdavies** [comment]
Exactly. The default chunking is every 30 seconds. My use-case is for the chunks to be used downstream for fine-tuning the model. Avoiding splits in the middle of words is an absolute must. Ensuring complete sentences are together is almost as important.

**Comment by nkilm** [comment]
> Exactly. The default chunking is every 30 seconds. My use-case is for the chunks to be used downstream for fine-tuning the model. Avoiding splits in the middle of words is an absolute must. Ensuring complete sentences are together is almost as important.

Ah got it. You can pass the `chunk_size` as param to [`TranscriptionConfig`](https://github.com/nkilm/offline-whisperx/blob/a07bbe5ac15b5e114fc448b2dcf864898e50a08a/transcriber.py#L30). Semantic chunking of audio is not  implemented afaik.

```diff
config = TranscriptionConfig(
    whisper_model_name=FASTER_WHISPER_PATH,  # provide whisper model name or path
    align_model_dir=ALIGN_MODEL_DIR,
    vad_model_fp=VAD_MODEL_FP,
    pyannote_config_path=PYANNOTE_CONFIG_PATH,
    compute_type="float8" if torch.cuda.is_available() else "int8",
+  chunk_size=20,
    diarize=True,
    output_dir="./output",
    output_format="json",
)
```
